### PR TITLE
fix cookie synchronization regression

### PR DESCRIPTION
**Pull request by LavrovArtem**

### CODE REVIEW

**Comment by LavrovArtem** [comment]
@testcafe-build-bot retest

File: src/client/sandbox/cookie/index.js [hunk 1]
```diff
@@ -221,6 +224,10 @@ export default class CookieSandbox extends SandboxBase {
     }
 
     syncWindowCookie (parsedCookies, win) {
+        // NOTE: Current function can be called before the 'attach' function was called
```
Comment (miherlosev) [inline src/client/sandbox/cookie/index.js]: // NOTE: This function can be called before the 'attach' call.

File: src/client/sandbox/cookie/index.js [hunk 2]
```diff
@@ -241,5 +248,16 @@ export default class CookieSandbox extends SandboxBase {
         super.attach(window);
 
         this.windowSync = new WindowSync(window, this, this.messageSandbox);
+
+        for (const { parsedCookies, win, resolve } of this.storedWindowSync) {
```
Comment (miherlosev) [inline src/client/sandbox/cookie/index.js]: Let's move this code to the 'syncCookieForNotAttachedWindows' function.

File: src/client/sandbox/cookie/index.js [hunk 3]
```diff
@@ -21,6 +22,8 @@ export default class CookieSandbox extends SandboxBase {
         this.messageSandbox = messageSandbox;
         this.cookieSync     = new CookieSync();
         this.windowSync     = null;
+
+        this.storedWindowSync = [];
```
Comment (miherlosev) [inline src/client/sandbox/cookie/index.js]: I propose to rename as `pendingWindowSync`

File: test/client/fixtures/sandbox/cookie-test.js [hunk 4]
```diff
@@ -357,6 +357,28 @@ test('actual cookie in iframe even if a synchronization message does not receive
         });
 });
 
+if (!browserUtils.isFirefox) {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/cookie-test.js]: Add comment, why we skip the test for Firefox browser

File: src/client/sandbox/cookie/index.js (hunk 2 above)
Comment (LavrovArtem) [reply src/client/sandbox/cookie/index.js]: what about `_processPendingWindowSync`?

**Comment by LavrovArtem** [comment]
@testcafe-build-bot retest

**Pull request merged**